v2/kompari/src/difference.rs

```diff
@@ -0,0 +1,72 @@
+use image::{Pixel, Rgba};
+
+use crate::Image;
+
+#[derive(Debug)]
+pub enum Difference {
+    None,
+    MissingFile,
+    LoadError,
+    SizeMismatch,
+    Content {
+        n_different_pixels: u64,
+        distance_sum: u64,
+        diff_image: Image,
+    },
+}
+
+pub fn compare_images(old: &Image, new: &Image) -> Difference {
+    if old.width() != new.width() || old.height() != new.height() {
+        return Difference::SizeMismatch;
+    }
+
+    let n_different_pixels: u64 = old
+        .pixels()
+        .zip(new.pixels())
+        .map(|(pl, pr)| if pl == pr { 0 } else { 1 })
+        .sum();
+
+    if n_different_pixels == 0 {
+        return Difference::None;
+    }
+
+    let mut distance_sum: u64 = 0;
+
+    let diff_image_data: Vec<u8> = old
+        .pixels()
+        .zip(new.pixels())
+        .flat_map(|(&p1, &p2)| {
+            let total_distance = pixel_distance(p1, p2);
+            distance_sum += total_distance;
+            if total_distance > 0 {
+                // Highlight the differences atop a semi-transparent view of the shared parts.
```

Comment:
This behaviour is different. However, having seen your screenshots in #13, I think the +green -red is nice.

However, I'd suggest combining the two, so that the diff image has context for the pixels which haven't changed.

In this PR, I think it would be best just to get the old version merged, rather than changing behaviour here. That is, the contents of this file can *largely* be dropped, although I do think the new function signature (without file system ops) is cleaner.